runtime/JavaScript/src/antlr4/TokenStreamRewriter.js

```diff
@@ -0,0 +1,340 @@
+import Token from "./Token.js";
+import Interval from "./misc/Interval.js";
+
+export default class TokenStreamRewriter {
+    // eslint-disable-next-line no-undef
+    static DEFAULT_PROGRAM_NAME = "default";
+
+    /**
+     * @param {import("./CommonTokenStream")} tokens 
+     */
```

Comment:
I now realize that the JSDoc declarations should probably be improved for this class. I'll put some effort into adding more of them.